library/common/src/main/java/com/google/android/exoplayer2/util/NalUnitUtil.java

```diff
@@ -252,17 +292,16 @@ public static int getH265NalUnitType(byte[] data, int offset) {
   }
 
   /**
-   * Parses an SPS NAL unit using the syntax defined in ITU-T Recommendation H.264 (2013) subsection
-   * 7.3.2.1.1.
+   * Parses a SPS NAL unit payload using the syntax defined in ITU-T Recommendation H.264 (2013)
+   * subsection 7.3.2.1.1.
    *
    * @param nalData A buffer containing escaped SPS data.
-   * @param nalOffset The offset of the NAL unit header in {@code nalData}.
+   * @param nalOffset The offset of the NAL unit in {@code nalData}.
    * @param nalLimit The limit of the NAL unit in {@code nalData}.
    * @return A parsed representation of the SPS data.
    */
-  public static SpsData parseSpsNalUnit(byte[] nalData, int nalOffset, int nalLimit) {
+  public static SpsData parseSpsNalUnitPayload(byte[] nalData, int nalOffset, int nalLimit) {
```

Comment:
Is there a strong reason for renaming this, given it's a breaking API change (which we can do if we need to, but it doesn't really feel like it's worth the potential disruption in this case)?

If not, let's remove "Payload" from the method below as well, so they're consistent.

Reply:
I renamed the method because I changed its behaviour. It now does no longer skip the 8 header bits because this would prevent the re-use of this method in `FragmentedMp4Extractor` in the other PR.

Yes, that's a breaking change. But if you want, I could re-add a `parseSpsNalUnit` method which skips 8 bits and then calls `parseSpsNalUnitPayload`.

Same goes for `parsePpsNalUnitPayload` and `parseH265SpsNalUnitPayload`.

Reply:
Ah, I didn't spot that. It might be worth keeping
```
parseSpsNalUnit(...) {
  parseSpsNalUnitPayload(nalData, nalOffset + 1, nalLimit);
}
```